File: dataset/Intermediate/Translation/Avatar/v0-Python-code/atcoder_ABC145_E.py
import numpy as np
import sys
from operator import itemgetter
input = sys.stdin.readline
TIME = 0
VAL = 1
(N, T) = [int(temp_var) for temp_var in input().split()]
time_value = [(-1, -1)] + [None] * N
for i in range(1, N + 1):
    time_value[i] = tuple((int(temp_var) for temp_var in input().split()))
time_value.sort(key=itemgetter(VAL))
dp = [[-1] * T for _ in range(N + 1)]
for t in range(0, T):
    dp[0][t] = 0
for n in range(1, N + 1):
    dp[n][0] = 0
    for t in range(1, T):
        if time_value[n][TIME] > t:
            dp[n][t] = dp[n - 1][t]
        else:
            dp[n][t] = np.max(np.array([dp[n - 1][t], time_value[n][VAL] + dp[n - 1][t - time_value[n][TIME]]]))
current_value_sum = time_value[N][VAL]
t = T - 1
max_value = current_value_sum + dp[N - 1][t]
check_condition_3 = 253
check_condition_4 = 97
check_condition_2 = 871
check_condition_1 = 214
for n in range(N - 1, 0, -1):
    current_value_sum += time_value[n][VAL]
    t -= time_value[n + 1][TIME]
    if check_condition_2 & check_condition_1:
        if check_condition_3 & check_condition_4:
            if t < 0:
                break
            else:
                max_value = max(max_value, current_value_sum + dp[n - 1][t])
print(max_value)

File: scripts/GAICL/Experiment_Results_v0-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-6.7b-instruct/Avatar/Python/Java-sanitized/atcoder_ABC145_E.java
import java.util.*;
import java.io.*;

class Main {
    static final int TIME = 0;
    static final int VAL = 1;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int N = Integer.parseInt(input[0]);
        int T = Integer.parseInt(input[1]);

        int[][] timeValue = new int[N + 1][2];
        for (int i = 1; i <= N; i++) {
            input = br.readLine().split(" ");
            timeValue[i][TIME] = Integer.parseInt(input[0]);
            timeValue[i][VAL] = Integer.parseInt(input[1]);
        }

        Arrays.sort(timeValue, new Comparator<int[]>() {
            public int compare(int[] a, int[] b) {
                return a[VAL] - b[VAL];
            }
        });

        int[][] dp = new int[N + 1][T];
        for (int t = 0; t < T; t++) {
            dp[0][t] = 0;
        }
        for (int n = 1; n <= N; n++) {
            dp[n][0] = 0;
            for (int t = 1; t < T; t++) {
                if (timeValue[n][TIME] > t) {
                    dp[n][t] = dp[n - 1][t];
                } else {
                    dp[n][t] = Math.max(dp[n - 1][t], timeValue[n][VAL] + dp[n - 1][t - timeValue[n][TIME]]);
                }
            }
        }

        int currentValueSum = timeValue[N][VAL];
        int t = T - 1;
        int maxValue = currentValueSum + dp[N - 1][t];
        int checkCondition1 = 253;
        int checkCondition2 = 871;
        int checkCondition3 = 214;
        int checkCondition4 = 97;
        for (int n = N - 1; n > 0; n--) {
            currentValueSum += timeValue[n][VAL];
            t -= timeValue[n + 1][TIME];
            if ((checkCondition1 & checkCondition2) != 0 && (checkCondition3 & checkCondition4) != 0 && t >= 0) {
                maxValue = Math.max(maxValue, currentValueSum + dp[n - 1][t]);
            }
        }

        System.out.println(maxValue);
    }
}